File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/Domain/Enrollment/Projections/EnrollmentList/EnrollmentListItem.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace CreditCardEnrollment.Domain.Enrollment.Projections.EnrollmentList;

public class EnrollmentListItem
{
    [BsonId]
    public required string Id { get; set; }
    
    public required string UserId { get; set; }
    
    public required string ProductId { get; set; }
    
    public required string Status { get; set; }
    
    public required string StatusReason { get; set; }
    
    public DateTime RequestedDate { get; set; }
    
    public DateTime? ReviewedOn { get; set; }
}

public class ProductName
{
    [BsonId]
    public required string Id { get; set; }
    
    public required string Name { get; set; }
}

File: application/backend-credit-card-enrollment/backend-typescript/src/creditCard/enrollment/projection/enrollmentList/EnrollmentListItem.ts
export class EnrollmentListItem {
    constructor(
        // needs to be _id to be recognized as an _id field by MongoDB
        public readonly _id: string,
        public readonly userId: string,
        public readonly productId: string,
        public readonly productName: string,
        public readonly requestedDate: Date,
        public readonly status: string,
        public readonly statusReason?: string,
        public readonly reviewedOn?: Date
    ) {}
}